pkg/config/expand.go

```diff
@@ -78,7 +78,9 @@ func (dc *DeploymentConfig) expand() {
 			"failed to apply deployment variables in modules when expanding the config: %v",
 			err)
 	}
-	dc.expandVariables()
+	if err := dc.expandVariables(); err != nil {
```

Comment:
expandVariables now returns an error, instead of log.Fatal on its own. Do you believe there should now be an TestExpandVariables unit test? If so, please create task for it in the backlog.

The coverage for hpc-toolkit/pkg/config appears to improve by 0.2% with this commit, so I do not believe the lack of a unit test should block the PR.